packages/canvas/canvas-sprite/src/CanvasSpriteRenderer.ts

```diff
@@ -76,9 +80,9 @@ export class CanvasSpriteRenderer
         // If smoothingEnabled is supported and we need to change the smoothing property for sprite texture
         const smoothingEnabled = texture.baseTexture.scaleMode === SCALE_MODES.LINEAR;
 
-        if (renderer.smoothProperty && renderer.context[renderer.smoothProperty] !== smoothingEnabled)
+        if (renderer.smoothProperty && (renderer.context as any)[renderer.smoothProperty] !== smoothingEnabled)
```

Comment:
What is `context` here? Can we avoid any?

Reply:
`context` is the `ContextSystem`. I believe it is set to `any` as we are trying to access a property dynamically. I never found a good way of doing this is TypeScript. 

This sort of thing is littered around the code base

If anyone knows how to fix this sort of error then we can tidy up all occurrences of this
```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'CanvasRenderingContext2D'.
  No index signature with a parameter of type 'string' was found on type 'CanvasRenderingContext2D'.
```

Reply:
Instead of using a string you and do `keyof SomeType` which makes it possible to index like this.

Reply:
@bigtimebuddy Cool idea, thanks, but in the next case I need to cast `context` to `any` 

```ts
(context as any)[renderer.smoothProperty as imageSmoothingKey] = smoothingEnabled;
```
because in another way I will get an error `TS2540: Cannot assign to 'canvas' because it is a read-only property.`

Reply:
There are [TypeScript utilities](https://www.typescriptlang.org/docs/handbook/utility-types.html) for dealing with this. So you could do something like `keyof Omit<SomeClass, 'canvas'>` to exclude certain properties (blacklist)  or `keyof Pick<SomeClass, 'smoothing'>` (whitelist).

Reply:
@bigtimebuddy fixed, thanks a lot for your help)